### ISSUE TITLE
Alerts. Can't filter out errors in the "IF" block

### ISSUE THREAD

**Issue by zzzkasper-1995** [getsentry/sentry#68393]
### Environment

self-hosted (https://develop.sentry.dev/self-hosted/)

### Steps to Reproduce

I'm trying to set up notifications so that they only fire on fatal errors and if they happen more than N times

I know for sure they are there because I get alerts if I remove everything from the if block.

I also launched two types of alerts in parallel, with and without a filter, and alerts with the level=fatal filter do not work.

Steps:
1) Click "Create Alert" on the Alert List screen.
2) Screen – “Select alert”, select the notification type “Issuess”.
3) In the WHEN block, select "Number of events in an issue is more than 1 in 30 days"
4) Make sure the preview shows errors with level = fatal.
5) in the iF block, select "The event's level is equal to fatal"
6) It is expected that only errors with a fatal level will show in the preview. Actual result, preview is empty.

PS: 
If I remove the IF block, then I receive error notifications with a fatal level. 
On the error list screen, I can successfully filter out fatal errors using the filter: "level:fatal"

Envirement: 
self-host
react-native: 0.71
@sentry/react-native: 5.19.1

<img width="868" alt="2024-04-02_10 37 22" src="https://github.com/getsentry/sentry/assets/25958211/a43dcdae-5a88-4057-963d-fba484b3f0c5">
<img width="789" alt="2024-04-02_10 36 13" src="https://github.com/getsentry/sentry/assets/25958211/de4861ec-f4b2-4607-8260-f6dcccdb6acc">

### Expected Result

The alert system sends messages with information about errors with level = fatal

### Actual Result

The alert system does NOT send messages with information about errors with level=fatal

### Product Area

Alerts

**Comment by mentaman** [comment]
Another use case:
We want to have alert over issues that got degradation in one region.
We want to get an alert over this issue.

So we've tried setting when: 5k users a day, if region = AU.

But the "if" condition works only over the last event, instead of filtering 5k users, which is quite an unexpected behaviour imo and less intuitive to understand. Though the biggest issue is that we can't have alerts over our regions, which would help us avoiding progressing faulty deploys to other regions.

**Comment by scttcper** [comment]
@mentaman i believe your ask is separate from @zzzkasper-1995 who's issue seems to be with the preview which currently doesn't support every combination of filter.

@mentaman issue alerts filters don't currently work together as a query builder, each filter is independent of each other. Metric alerts might be the closest you could get or if your region happens to be in the environment this would also apply to the 5k users in interval query.
![image](https://github.com/getsentry/sentry/assets/1400464/e1814bb9-e051-41eb-abce-542020b7533b)

**Comment by xiefeitong01** [comment]
Has this issue been solved? I also encountered this problem on my self-hosted sentry. Not only can I not preview it, but it actually cannot function normally when trigger alert.

**Comment by leedongwei** [comment]
@zzzkasper-1995 @xiefeitong01 Do you still see the same problem? I'm not able to replicate it on my end.

**Comment by AntonisAgap** [comment]
Hello! Has there been any update with this bug? We're having the same issue, where the filter doesn't work. 
We tried filtering by the:
`event.message` and `event.tags` but it doesn't seem to work.

**Comment by rachrwang** [comment]
@AntonisAgap - can you share the full configuration of your alert, including the trigger and filters?

**Comment by bnawras** [comment]
@rachrwang @leedongwei 

I'm experiencing the same issue. I receive an alert even when the message doesn't contain the specified word. Interestingly, if I change the word to something random, it seems to work correctly. It looks like the specified word appears in the stack trace, file name, which might be triggering the alert.

![Image](https://github.com/user-attachments/assets/858f258f-73b4-4b95-88ee-1da36f4d457c)

**Comment by bnawras** [comment]
I expect Sentry to check for the word only in the message
![Image](https://github.com/user-attachments/assets/813b02c1-5edb-4dff-896d-22be9ecf6ddb)

This message contains 'provided', but there are other issues without any form of this word in the message

**Comment by kcons** [comment]
@bnawras What is the intent of the "Number of events in an issue is more than 0 in one minute" clause? It's already being triggered by a new issue there, so this is the first case of it. Not that this should impact attribute comparison filters (those are all fairly simple and seem to work in the environments I have access to), but adding a event frequency clause does change how alerts are executed a bit, so I'm curious if removing that has any consequence.

**Comment by bnawras** [comment]
@kcons Thanks for noticing that. The first trigger, “a new issue is created,” isn’t needed. I want to receive alerts for every event related to a specific issue. A workaround for the problem, as explained above, is to use a more specific phrase that doesn’t appear in the stack trace. For example, I replaced "provider" with "All providers" and specific provider names, and that works for me.

**Comment by kcons** [comment]
If you're finding that alerts aren't firing correctly based on the 'when' and 'if' criteria you sent, I'd recommend filing a new task with the details there; this task is about the preview queries, which are known to be incomplete.
We're currently working on a new version of the infrastructure and UI involved here, so we're not too focused on fixing non-critical issues with the current UI; with that, and the fact that this task is largely attracting comments on separate issues, I'll close this for now.
